python_modules/libraries/dagster-sling/dagster_sling_tests/code_locations/sling_location/defs/singular_ingest/component.yaml

```diff
@@ -0,0 +1,13 @@
+type: dagster_sling.SlingReplicationComponent
+
+attributes:
+  replication:
+    path: "./replication.yaml"
+    asset_attributes:
+      key: "singular/{{ stream_definition.config.meta.dagster.asset_key }}"
+  sling:
+    connections:
```

Comment:
One obvious downside to this change is that a user must repeat their connections in every component document. Maybe providers solve this, and all configuration of this sort ends up pushed into the resource system?

Reply:
yeah given that this part of the yaml is just used to instantiate a resource, definitely feels like pushing this to providers / the resource system feels like the right end state